### PR TITLE
utf-8 decode error

### LINKED ISSUE #111: codec can't decode byte
during startup my esp32 board returns (sometimes) crazy data causing mpfshell to fail

```
Traceback (most recent call last):
  File "/home/xx/repo/mpfshell/mpfshell.py", line 5, in <module>
    sys.exit(main())
  File "/home/xx/repo/mpfshell/mp/mpfshell.py", line 914, in main
    mpfs.onecmd(scmd)
  File "/usr/lib/python3.10/cmd.py", line 217, in onecmd
    return func(arg)
  File "/home/xx/repo/mpfshell/mp/mpfshell.py", line 564, in do_exec
    ret = self.fe.follow(None, data_consumer)
  File "/home/xx/repo/mpfshell/mp/pyboard.py", line 105, in follow
    data = self.read_until(1, b"\x04", timeout=timeout, data_consumer=data_consumer)
  File "/home/xx/repo/mpfshell/mp/pyboard.py", line 48, in read_until
    data_consumer(new_data)
  File "/home/xx/repo/mpfshell/mp/mpfshell.py", line 556, in data_consumer
    data = str(data.decode("utf-8"))
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc9 in position 0: unexpected end of data

```

https://github.com/wendlers/mpfshell/blob/3ac8f401077971b2026e51f2c2180bd7b71157a5/mp/mpfshell.py#L554-L558

### PR DESCRIPTION
**Pull request by kr-g**
fix #111

https://github.com/wendlers/mpfshell/issues/111

### CODE REVIEW

**Comment by kr-g** [comment]
this will change the output to an escaped sequence of bytes preluded by "\x"

**Comment by skorokithakis** [comment]
This looks good to me, but can you add a comment detailing why this is necessary? It's a bit of a mystery right now to anyone reading the code.

**Comment by kr-g** [comment]
in fact i m not so happy with my own proposal.
so now its just a workaround.
i will have a look at it again this weekend when i have more time.

File: mp/mpfshell.py [hunk 1]
```diff
@@ -554,11 +554,11 @@ def do_exec(self, args):
...
-            try:
-                data = str(data.decode("utf-8"))
-            except UnicodeDecodeError:
-                data = "\\x" + binascii.hexlify(data).decode()
-            sys.stdout.write(data.strip("\x04"))
+            data = data.replace(b"\x04", b"")
+            # write raw bytes to stdout
+            sys.stdout.buffer.write(data)
+            # make sure that no data remains in an internal buffer
+            sys.stdout.flush()
```
Comment (skorokithakis) [inline mp/mpfshell.py:561]:
```suggestion
            # Delete garbage characters, as they can make the connection fail.
            data = data.replace(b"\x04", b"")
            sys.stdout.buffer.write(data)
            sys.stdout.flush()
```

**Pull request merged**